src/sap/fhir/model/r4/FHIRUtils.js

```diff
@@ -497,14 +503,61 @@ sap.ui.define([
 		} else if (oFilter.sOperator === FHIRFilterOperator.BT) {
 			var oValue2 = FHIRFilterOperatorUtils.getFilterValue(oFilter.oValue2);
 			mParameters[sPath + sFhirSearchModifier] = ["ge" + oValue1, "le" + oValue2];
-		} else if (bLogicalConnection && !bLogicalOperator){
-			if (mParameters[sPath + sFhirSearchModifier]){
+		} else if (bLogicalConnection && !bLogicalOperator) {
+			if (mParameters[sPath + sFhirSearchModifier]) {
 				mParameters[sPath + sFhirSearchModifier].push(vValue);
 			} else {
-				mParameters[sPath + sFhirSearchModifier] = [ vValue ];
+				mParameters[sPath + sFhirSearchModifier] = [vValue];
 			}
 		} else {
-			mParameters[sPath + sFhirSearchModifier] =  vValue;
+			mParameters[sPath + sFhirSearchModifier] = vValue;
+		}
+	};
+
+	/**
+	 * Creates a complex filter
+	 *
+	 * @param {object} oFilter The filter which should be added to the parameters
+	 * @param {object} mParameters The parameters which should be passed to the request
+	 * @param {string} sLogicalConnection if the list of filters needs to be combined either with AND or OR
+	 * @private
+	 * @since 2.1.0
+	 */
+	FHIRUtils._complexFilterBuilder = function (oFilter, mParameters, sLogicalConnection) {
+		var sLogicalConnection1;
+		if (oFilter instanceof Filter) {
+			if (oFilter._bMultiFilter) {
+				// recursive
+				sLogicalConnection1 = oFilter.bAnd && oFilter.bAnd == true ? " and " : " or ";
+				if (oFilter.aFilters) {
+					mParameters._filter = mParameters._filter + "( ";
+					// for the first filter the logical connection shouldnt be appended
+					this._complexFilterBuilder(oFilter.aFilters[0], mParameters, undefined);
+					for (var i = 1; i < oFilter.aFilters.length; i++) {
+						this._complexFilterBuilder(oFilter.aFilters[i], mParameters, sLogicalConnection1);
+					}
+					mParameters._filter = mParameters._filter + " )";
+				}
+			} else {
+				// validate the filter operator
+				// if BT operator use 'and' to generate the filter value
+				var sPath = oFilter.sPath;
+				var sFilterOperator = FHIRFilterOperatorUtils.getFHIRFilterPrefix(oFilter);
+				var oValue1 = FHIRFilterOperatorUtils.getFilterValueForComplexFilter(oFilter.sValueType, oFilter.oValue1);
+				var oValue2;
+				var sFilter;
+				if (oFilter.sOperator === FHIRFilterComplexOperator.BT) {
+					oValue2 = FHIRFilterOperatorUtils.getFilterValueForComplexFilter(oFilter.sValueType, oFilter.oValue2);
+					sFilter = "( " + sPath + " ge " + oValue1 + " and " + sPath + " le " + oValue2 + " )";
```

Comment:
i would prefer using `FHIRFilterComplexOperator.GT` instead of "ge", same for "le"

Reply:
as per discussion